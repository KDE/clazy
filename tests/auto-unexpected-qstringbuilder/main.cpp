#define QT_USE_QSTRINGBUILDER
#include <QtCore/QString>

void test()
{
    const auto s1 = "tests/" + QString::fromLatin1("foo"); // Warning
    const auto s2 = "tests/" % QString::fromLatin1("foo"); // Warning
    const QString s3 = "tests/" + QString::fromLatin1("foo"); // OK
    const QString s4 = "tests/" % QString::fromLatin1("foo"); // OK

    [] {
        return "tests/" % QString::fromLatin1("foo"); // Warn
    };

    [] {
        return QString(); // OK
    };
}

