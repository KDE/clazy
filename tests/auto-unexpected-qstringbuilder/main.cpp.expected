auto-unexpected-qstringbuilder/main.cpp:6:5: warning: auto deduced to be QStringBuilder instead of QString. Possible crash. [-Wclazy-auto-unexpected-qstringbuilder]
auto-unexpected-qstringbuilder/main.cpp:7:5: warning: auto deduced to be QStringBuilder instead of QString. Possible crash. [-Wclazy-auto-unexpected-qstringbuilder]
auto-unexpected-qstringbuilder/main.cpp:11:5: warning: lambda return type deduced to be QStringBuilder instead of QString. Possible crash. [-Wclazy-auto-unexpected-qstringbuilder]
